# Inferencing

# 1. Import Library

Yang pertama adalah import library untuk pemrosesan data

In [1]:
# Import Libraries
import pickle
import json
import pandas as pd
import numpy as np

# 2. Import File Model

Import file model dengan menggunakan bantuan library `json` dan `pickle`

In [2]:
#Import file model
with open('censoring.pkl', 'rb') as file_1:
  windsoriser = pickle.load(file_1)

with open('norm_scaler.pkl', 'rb') as file_2:
  norm_scaler = pickle.load(file_2)

with open('skew_scaler.pkl', 'rb') as file_3:
  skew_scaler = pickle.load(file_3)

with open('encoder.pkl', 'rb') as file_4:
  encoder= pickle.load(file_4)

with open('mod_ridge_opt.pkl', 'rb') as file_5:
  mod_ridge_opt = pickle.load(file_5)

with open('num_columns_train.txt', 'r') as file_6:
  num_columns_train= json.load(file_6)

with open('cat_columns_train.txt', 'r') as file_7:
  cat_columns_train= json.load(file_7)

with open('norm_feature.txt', 'r') as file_8:
  norm_feature= json.load(file_8)

with open('skew_feature.txt', 'r') as file_9:
  skew_feature= json.load(file_9)


# 3. Membuat Data Inferencing

Saya membuat data 1 Data Inferencing dan convert ke dataframe dengan bantuan pandas

In [3]:
# Create New Data 

data_inf = {
    'cab_type': 'Uber', 
    'name': 'UberXL', 
    'distance': 3, 
    'surge_multiplier': 1.1, 
}

data_inf = pd.DataFrame([data_inf])
data_inf

cab_type    name  distance  surge_multiplier
0     Uber  UberXL         3               1.1

# 4. Windsoriser Data Inferencing

Kemudian dilakukan windsoriser pada Data Inferencing

In [4]:
#Windsoriser pada data inferencing dengan modul windsoriser
data_inf = windsoriser.transform(data_inf)

# 5. Scaling dan Encoding

Selanjutnya adalah scaling dan encoding pada feature. Sebelumnya adalah pemisahan kolom number dan categorical

In [5]:
#Pemisahan kolom numerical dan categorical
data_inf_num = data_inf[num_columns_train]
data_inf_cat = data_inf[cat_columns_train]

data_inf_num

distance  surge_multiplier
0         3               1.1

Selanjutnya adalah melakukan scaling pada feature normal dan feature skew

In [6]:
#Scaling data skew dan normal
data_inf_num_scaled = data_inf_num.copy()
data_inf_num_scaled[norm_feature] = norm_scaler.transform(data_inf_num[norm_feature])
data_inf_num_scaled[skew_feature] = skew_scaler.transform(data_inf_num[skew_feature])

In [7]:
#Pengecekan apakah berhasil scaling atau tidak
data_inf_num_scaled

distance  surge_multiplier
0  0.817253               0.1

Selanjutnya adalah encoding fitur categorical

In [8]:
#encoding feature categorical
data_inf_cat_encoded = pd.DataFrame(encoder.transform(data_inf_cat).toarray())

In [9]:
#Pengecekan apakah berhasil encoding atau tidak

Selanjutnya adalah concate semua feature sebelum melakukan prediksi

In [10]:
#Concate all feature
data_inf_final = np.concatenate([data_inf_num_scaled, data_inf_cat_encoded], axis=1)

# 6. Predict Data Inferencing

Kemudian adalah prediksi data menggunakan model Linear Regression Optimization

In [11]:
# Predict using Linear Regression

y_pred_inf = mod_ridge_opt.predict(data_inf_final)
y_pred_inf

array([20.40489905])

Harga untuk perjalanan **3 km** dengan layanan **Uber** dan **fasilitas** `UberXL` dengan tingkat demand **1.1** adalah **20.4**